server/src/main/java/org/elasticsearch/action/ActionListenerImplementations.java

```diff
@@ -254,6 +255,33 @@ public String toString() {
         }
     }
 
+    /**
+     * The same as {@link ResponseWrappingActionListener} except that the response is dropped
+     */
+    static final class ResponseDroppingActionListener<T, R> extends DelegatingActionListener<T, R> {
```

Comment:
Making a specialized class here instead of wrapping the `CheckedConsumer` in a drop-the-arg lambda, which avoids an unnecessary allocation and a layer of indirection.